### PR TITLE
upd readme[duda]

### PR DESCRIPTION
**Pull request by Dmitry-Duda**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -4,20 +4,30 @@
...
 <!-- run online end -->
 
+## Files to Look At
+
+* [Default.aspx](./CS/WebSite/Default.aspx) (VB: [Default.aspx](./VB/WebSite/Default.aspx))
+* [Default.aspx.cs](./CS/WebSite/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/WebSite/Default.aspx.vb))
+
+## Implementation Details
+
+In this example, [`ASPxGridView`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) uses a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object as its data source. To bind the grid to the [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object, the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event handler assigns this table to the grid's [`DataSource`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxDataWebControlBase.DataSource) property.
```
Comment (almikholap) [inline Readme.md]: ~"#ConciseText_Style" Just making the sentences more straightforward... Note that I simplified the second paragraph too, so here's the complete section:

```suggestion
In this example, the grid's [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event handler assigns a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object to the grid's [`DataSource`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxDataWebControlBase.DataSource) property. To persist the object between [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event occurrences, the example stores it in [session state](https://docs.microsoft.com/en-us/previous-versions/aspnet/ms178581(v=vs.100)).

File: Readme.md [hunk 2]
```diff
@@ -4,20 +4,30 @@
...
+* [Default.aspx.cs](./CS/WebSite/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/WebSite/Default.aspx.vb))
+
+## Implementation Details
+
+In this example, [`ASPxGridView`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) uses a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object as its data source. To bind the grid to the [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object, the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event handler assigns this table to the grid's [`DataSource`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxDataWebControlBase.DataSource) property. 
+
+The example stores the initially created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object in [session state](https://docs.microsoft.com/en-us/previous-versions/aspnet/ms178581(v=vs.100)). This technique allows you not to recreate the table for subsequent [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event triggers.
```
Comment (almikholap) [inline Readme.md]: ~"#Wordiness_SentenceStructure" I'd remove `initially` as it doesn't add anything to the sentence - we're just referring to the same object we created.

~"#TermUsage_TechnicalSide" As for `triggers`, are you referring to the things that trigger/raise the event (ASPxGridView1.DataBind();) or event occurrences (so we use session state and don't recreate the object on each event occurrence)? Please clarify. I based my suggestion on the latter.

> for subsequent [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event **triggers**

Comment (Dmitry-Duda) [reply Readme.md]: I've used `initially` to focus a user's attention on the fact that the table is created only once and isn't recreated later. So your suggestion `To persist the object between...` is technically correct, but I've tried to emphasize the lack of the recreate action for better performance. In my opinion, this is the main idea of keeping the table in session state in this example.
```
DataTable GetTable() {
  //You can store a DataTable in the session state
  DataTable table = Session["Table"] as DataTable;
  if (table == null) {
      table = new DataTable();
      table.Columns.Add("id", typeof(int));
      table.Columns.Add("data", typeof(String));
      for (int n = 0; n < 100; n++) {
          table.Rows.Add(n, "row" + n.ToString());
  }
  Session["Table"] = table;
}
```
What do you think about it?


Regarding `triggers`, your second option is what I meant - occurrences. Is `triggers` an inappropriate term?

Comment (almikholap) [reply Readme.md]: I just suggest to omit `initially` and see if anything changes when you read the sentence. Do we need this adverb as a modifier to `created`? I think that without the modifier the sentence reads clearer, as if we're referring to the same (created) object.

> In my opinion, this is the main idea of keeping the table in session state in this example.

I see your point regarding the main idea. How about this addition? `This technique allows you not to recreate the object each time the DataBinding event fires.`

> Regarding `triggers`, your second option is what I meant - occurrences. Is `triggers` an inappropriate term?

Stick with the `occurrence` term then - as I described above, an event trigger is what triggers/raises an event, not its occurrence.

Comment (Dmitry-Duda) [reply Readme.md]: > I think that without the modifier the sentence reads clearer, as if we're referring to the same (created) object.

Okay, along with the mention that we are "preserving this object between `DataBinding` event occurrences", I agree that this is clear enough

---

> I see your point regarding the main idea. How about this **addition**? `This technique allows you not to recreate the object each time the DataBinding event fires.`

Do you suggest adding the above sentence after the following part?   
> To persist the object between `DataBinding` event occurrences, the example stores it in **session state**.

In this example, the grid's `DataBinding` event handler assigns a programmatically created `DataTable` object to the grid's `DataSource` property. To persist the object between `DataBinding` event occurrences, the example stores it in **session state**. This technique allows you not to recreate the object each time the `DataBinding` event fires.

---

> Stick with the occurrence term then - as I described above, an event trigger is what triggers/raises an event, not its occurrence.

I see, thank you!

Comment (almikholap) [reply Readme.md]: > Do you suggest adding the above sentence after the following part?

Exactly.

Comment (Dmitry-Duda) [reply Readme.md]: Okay, thanks! I'll make the changes.

File: CS/WebSite/web.config [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
+      <section name="themes" type="DevExpress.Web.ThemesConfigurationSection, DevExpress.Web.v21.2, Version=21.2.4.0, Culture=neutral, PublicKeyToken=b88d1754d700e49a" requirePermission="false" />
+      <section name="compression" type="DevExpress.Web.CompressionConfigurationSection, DevExpress.Web.v21.2, Version=21.2.4.0, Culture=neutral, PublicKeyToken=b88d1754d700e49a" requirePermission="false" />
+      <section name="settings" type="DevExpress.Web.SettingsConfigurationSection, DevExpress.Web.v21.2, Version=21.2.4.0, Culture=neutral, PublicKeyToken=b88d1754d700e49a" requirePermission="false" />
+      <section name="errors" type="DevExpress.Web.ErrorsConfigurationSection, DevExpress.Web.v21.2, Version=21.2.4.0, Culture=neutral, PublicKeyToken=b88d1754d700e49a" requirePermission="false" />
+      <section name="resources" type="DevExpress.Web.ResourcesConfigurationSection, DevExpress.Web.v21.2, Version=21.2.4.0, Culture=neutral, PublicKeyToken=b88d1754d700e49a" requirePermission="false" />
     </sectionGroup>
   </configSections>
   <!--
```
Comment (albertov05) [inline CS/WebSite/web.config:12]: I have a few comments regarding line 15 below.

Line 15:
`The following attributes can be set on the <httpRuntime> tag.`

1. Is "`on the <httpRuntime> tag`" correct? Or do you set the attributes _within_ the tags?
    If so, would it be better to say, "`in the <httpRuntime> tag`"?

2. Please change the period at the end to a colon.

File: Readme.md [hunk 4]
```diff
@@ -4,20 +4,28 @@
...
 <!-- run online end -->
 
+## Files to Look At
+
+* [Default.aspx](./CS/WebSite/Default.aspx) (VB: [Default.aspx](./VB/WebSite/Default.aspx))
+* [Default.aspx.cs](./CS/WebSite/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/WebSite/Default.aspx.vb))
+
+## Implementation Details
+
+In this example, the grid's [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event handler assigns a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object to the grid's [`DataSource`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxDataWebControlBase.DataSource) property. To persist the object between [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event occurrences, the example stores it in [session state](https://docs.microsoft.com/en-us/previous-versions/aspnet/ms178581(v=vs.100)). This technique allows you not to recreate the object each time the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event fires.
```
Comment (albertov05) [inline Readme.md]: `This technique allows you not to recreate the object each time the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event fires.`

When you say "not to recreate the object each time", do you mean that the technique _prevents_ you from recreating the object each time?
Or do you mean that you _don't have to recreate_ the object each time? If you mean the latter, can we say something like the following?

`With this technique, you do not have to recreate the object each time the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event fires.`

File: CS/WebSite/web.config (hunk 3 above)
Comment (Dmitry-Duda) [reply CS/WebSite/web.config:12]: Okay, I agree with you. Thank you!

File: Readme.md (hunk 4 above)
Comment (Dmitry-Duda) [reply Readme.md]: Your second suggestion sounds good, thank you!
```suggestion
In this example, the grid's [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event handler assigns a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object to the grid's [`DataSource`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxDataWebControlBase.DataSource) property. To persist the object between [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event occurrences, the example stores it in [session state](https://docs.microsoft.com/en-us/previous-versions/aspnet/ms178581(v=vs.100)). With this technique, you do not have to recreate the object each time the [DataBinding](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event fires.
```

File: CS/WebSite/web.config (hunk 3 above)
Comment (Dmitry-Duda) [reply CS/WebSite/web.config:12]: https://github.com/DevExpress-Examples/bind-a-grid-to-a-datatable-via-code-e168/pull/4/commits/3d1b9216fc388cb0c4d673d123c88e089358cfea

File: Readme.md (hunk 4 above)
Comment (albertov05) [reply Readme.md]: Thank you!

File: CS/WebSite/web.config (hunk 3 above)
Comment (albertov05) [reply CS/WebSite/web.config:12]: Thanks!

**Pull request merged**